### PR TITLE
Remove inspector flag

### PR DESCRIPTION
**Pull request by huwshimi**
Remove the remaining flags for the inspector, charmbrowser and env switcher. There will be a followup to actually remove the old code.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- view the gui without any flags
- everything should behave as if the react flag were applied

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3401/
Test FAILed.

File: jujugui/static/gui/src/app/views/state.js [hunk 1]
```diff
@@ -275,13 +271,6 @@ YUI.add('juju-app-state', function(Y) {
         if (id) {
           id = id.replace(/~charmers\//, '');
         }
-        // Setup the search status and filters based on metadata.search
-        if (!window.flags || !window.flags.react) {
-          if (Y.Lang.isValue(metadata.search)) {
-            search = true;
-            this.filter.update(metadata.search);
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/state.js]: I don't think that `this.filter` (and by extension the class on l780) is used any more, along with the `_addQueryState` method now that the changes on l345 are the default

File: jujugui/static/gui/src/app/utils/environment-change-set.js [hunk 2]
```diff
@@ -819,12 +819,7 @@ YUI.add('environment-change-set', function(Y) {
       // new React configuration system the modified values is determined
       // in the view and set in the service model so we can faithfully
       // take what it says to set as correct.
-      var changedFields;
-      if (!window.flags || !window.flags.react) {
-        changedFields = utils.getChangedConfigOptions(config, serviceConfig);
```
Comment (hatched) [inline jujugui/static/gui/src/app/utils/environment-change-set.js]: I don't think that the `getChangedConfigOptions` method is used anywhere else.

File: jujugui/static/gui/src/app/subapps/browser/browser.js [hunk 3]
```diff
@@ -740,9 +738,6 @@ YUI.add('subapp-browser', function(Y) {
       // so this method only needs these lines once switched over.
       // We need to render the sidebar view as default. This is the new design
       // in the near future we will likely just render it in the initializer.
-      if (!window.flags || !window.flags.react) {
-        this.sidebar();
```
Comment (hatched) [inline jujugui/static/gui/src/app/subapps/browser/browser.js]: Can you remove this method and by extension the Sidebar view, tests (anything it uses, etc)

File: jujugui/static/gui/src/app/store/env/go.js [hunk 4]
```diff
@@ -1787,11 +1787,7 @@ YUI.add('juju-env-go', function(Y) {
         // new React configuration system the modified values is determined
         // in the view and set in the service model so we can faithfully
         // take what it says to set as correct.
-        if (!window.flags || !window.flags.react) {
-          newValues = utils.getChangedConfigOptions(config, serviceConfig);
```
Comment (hatched) [inline jujugui/static/gui/src/app/store/env/go.js]: see about removing this below.

File: jujugui/static/gui/src/app/models/models.js [hunk 5]
```diff
@@ -244,20 +244,6 @@ YUI.add('juju-models', function(Y) {
       this.set(
           'environmentConfig',
           Y.merge(this.get('environmentConfig'), changeConfig));
-      // If there are any fields where the local version is different from the
-      // new environment fields that means that it has been changed locally.
-      // XXX We will want to enable conflicted field support for the react
-      // stuff in the future but for now it's only causing issues with the UI.
-      if (!window.flags || !window.flags.react) {
-        if (conflicted.length > 0) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/models/models.js]: If you're not using `conflicted` anymore then you can remove how it's generated above.

**Comment by hatched** [comment]
Ahh this is looking great. My comments are based around the idea that if you removed the only place that calls a block of code be sure to remove the now-dead code as well. We tend to leave things like that hanging around for a long time by accident.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3402/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3411/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3412/
Test PASSed.

**Comment by hatched** [comment]
:+1: QA :ok: WIth our discussion on IRC that you're going to keep track of those methods to remove and then do follow-up branches.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3413/
Test FAILed.

File: jujugui/static/gui/src/app/subapps/browser/browser.js (hunk 3 above)
Comment (makyo) [inline jujugui/static/gui/src/app/subapps/browser/browser.js]: Agreed, but maybe for this branch just a TODO comment

**Comment by makyo** [comment]
:+1:!

**Comment by huwshimi** [comment]
As discussed I'm removing the other parts in a followup branch. Thanks for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**